### ISSUE TITLE
SE: SVT Play don't display folder lists anymore

### ISSUE THREAD

**Issue by matsarv** [retrospect-addon/plugin.video.retrospect#1628]
### Describe the bug

SVT Play don't display folder lists, log says "process_folder_list returned 0 items"

## Expected Behavior

List program items

## Actual Behavior

Empty folder lists, nothing to play

### Debuglog

The **debug** retrospect.log can be found here: https://paste.kodi.tv/tidigopeba.kodi

### Your Environment
Used Operating system:

 - [X] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [X] Raspberry-Pi
 - [X] Windows
 - [ ] Windows UWP

Used versions:
- Kodi version: 19.4
- Retrospect version: 5.5.3

**Comment by basrieter** [comment]
Bits and bits are starting to work again. Test it with https://github.com/retrospect-addon/plugin.video.retrospect/archive/refs/heads/bug-1628.zip in Kodi Matrix.

**Comment by basrieter** [comment]
All should be fine again. Can somebody verify this. If not, please attach a debug file.

**Comment by matsarv** [comment]
Greate. I've tested all and found two things not working, gives "returned 0 items":

1. Live Tv
2. Search

New debug retrospect.log can be found here: https://paste.kodi.tv/iwahonazaw.kodi

**Comment by Sopor** [comment]
## Bug report
### Describe the bug



`Marika Carlsson är nyfiken döv` will not show any year or season, but it seems to be the same on [svtplay](https://www.svtplay.se/marika-carlsson-ar-nyfiken-dov)


## Expected Behavior





## Actual Behavior





## Possible Fix


I don't know if there are more like this, but maybe you could add some dummy text when this happens?


### To Reproduce
Steps to reproduce the behavior:



1. 
2.
3.

### Debuglog



The **debug** retrospect.log can be found here: https://paste.kodi.tv/yudetujifu.kodi


### Screenshots 

![image](https://user-images.githubusercontent.com/5789283/161829819-e140cf2f-ca50-4618-98a9-3018cd86a4b9.png)



## Additional context or screenshots (if appropriate)





### Your Environment
Used Operating system:


 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [x] Windows
 - [ ] Windows UWP

Used versions:
- Kodi version: 19.4
- Retrospect version: 5.5.4+bug-1628

**Comment by Sopor** [comment]
`Now playing` will not show any season and episode number.

https://paste.kodi.tv/toqifayefu.kodi

![image](https://user-images.githubusercontent.com/5789283/161832003-1d4d5eab-5d8e-4c7f-be29-6e3114c54014.png)

**Comment by Sopor** [comment]
I don't know how long these two have been visible, but is it possible to add an option to hide them?

Most users will never use them.

![image](https://user-images.githubusercontent.com/5789283/161832611-03cc02f0-bd13-4421-b8f1-9f15c42efbeb.png)

**Comment by matsarv** [comment]
> I don't know if there are more like this, but maybe you could add some dummy text when this happens?

I've added this for test, (my guess this is when season or year is missing):
![missing](https://user-images.githubusercontent.com/46671821/161843726-4736e6ff-19d8-4d11-b297-8c7925c7f7a7.jpg)
mediaitem.py
def get_kodi_item

```
Replace row 429
        name = self.__get_title(name)
With
        name = self.__get_title(name)
        if name == "":
            name = LanguageHelper.get_localized_string(LanguageHelper.Episodes)
```
There is probably a better way!

Live TV is now working OK.

**Comment by basrieter** [comment]
> I don't know how long time these two have been visible, but is it possible to add an option to hide them?
> 
> Most users will never use them.
> 
> You can use the log from the comment above.
> 
> ![image](https://user-images.githubusercontent.com/5789283/161832611-03cc02f0-bd13-4421-b8f1-9f15c42efbeb.png)

You can always hide specific folders via the context menu.

**Comment by basrieter** [comment]
> > I don't know if there are more like this, but maybe you could add some dummy text when this happens?
> 
> I've added this for test, (my guess this is when season or year is missing): ![missing](https://user-images.githubusercontent.com/46671821/161843726-4736e6ff-19d8-4d11-b297-8c7925c7f7a7.jpg) mediaitem.py def get_kodi_item
> 
> ```
> Replace row 429
>         name = self.__get_title(name)
> With
>         name = self.__get_title(name)
>         if name == "":
>             name = LanguageHelper.get_localized_string(LanguageHelper.Episodes)
> ```
> 
> There is probably a better way!
> 
> Live TV is now working OK.

Fixed now.

**Comment by basrieter** [comment]
> I don't know how long time these two have been visible, but is it possible to add an option to hide them?
> 
> Most users will never use them.
> 
> You can use the log from the comment above.
> 
> ![image](https://user-images.githubusercontent.com/5789283/161832611-03cc02f0-bd13-4421-b8f1-9f15c42efbeb.png)

Should be fixed now. Can you redownload and retest.

**Comment by Sopor** [comment]
> > I don't know how long time these two have been visible, but is it possible to add an option to hide them?
> > Most users will never use them.
> > You can use the log from the comment above.
> > ![image](https://user-images.githubusercontent.com/5789283/161832611-03cc02f0-bd13-4421-b8f1-9f15c42efbeb.png)
> 
> You can always hide specific folders via the context menu.

So, i need to hide these two folders for every TV program that have it?

**Comment by basrieter** [comment]
I don't think all have? What does it mean?

**Comment by matsarv** [comment]
> Fixed: Some folders will not show any name (See #1628).
> Fixed: Don't set item as episode, if no info is present (See #1628).

Checked the two fixes, working OK.

I found some minor changes in the "Categories", changed topics and images.
Added two more, shown as link on the web site but on the Android-app they are shown as images.

Changes: [category_items_changes.txt](https://github.com/retrospect-addon/plugin.video.retrospect/files/8424713/category_items_changes.txt)

**Comment by matsarv** [comment]
> I don't think all have? What does it mean?

If you have added the above category changes, you will see how many have:
- syntolkat = 81/2643 tv shows
- teckenspråkstolkat = 104/2643 tv shows

syntolkat = Visual interpretation is an aid mainly for people with visual impairment.
teckenspråkstolkat = Sign language interpretation is an aid mainly for  deaf or hard of hearing.

This is a great help for people who have the need. I do not think that these should just be hidden. If it is to be implemented, it should be done with a show / hide setting in SVT Play channel settings.

**Comment by basrieter** [comment]
> > I don't think all have? What does it mean?
> 
> If you have added the above category changes, you will see how many have:
> 
> * syntolkat = 81/2643 tv shows
> * teckenspråkstolkat = 104/2643 tv shows
> 
> syntolkat = Visual interpretation is an aid mainly for people with visual impairment. teckenspråkstolkat = Sign language interpretation is an aid mainly for deaf or hard of hearing.
> 
> This is a great help for people who have the need. I do not think that these should just be hidden. If it is to be implemented, it should be done with a show / hide setting in SVT Play channel settings.

Thanks for explaining what they mean. I am filtering them out now, but I will add an option later tonight.

**Comment by basrieter** [comment]
![image](https://user-images.githubusercontent.com/1310443/162071341-5fbf6b48-5dcd-4aef-b1f7-0a12b7fcc61b.png)

You can configure it now @Sopor  @matsarv

**Comment by matsarv** [comment]
Tested, works perfectly.
For test, I've added the swedish language text: "Visa program med syn och teckentolkat i mappar"

**Comment by basrieter** [comment]
I used "Visa program med syn och teckentolkat". Would that also do?

**Comment by matsarv** [comment]
Works for me.

**Comment by matsarv** [comment]
Found another bug. Some info is missing.

Under "Categories" or "Genres" the Description and Expires is missing.
1. Goto SVT Play
2. Select Categories or Genres
3. Select Serier
4. Goto Jamestown (shift+j) Show folder. "No info to the left" 
5. Select Jamestown and then mark an episode. "No info to the left"

Under "Recent" the Description and Expires are there.
1. Goto SVT Play
2. Select Recent
3. Goto Jamestown (shift+j) then 2 down. "All info is shown to the left"

I can make some images if you need!

 
Previous reported above:

- Search not working.
- Minor changes in the static "Categories". [category_items_changes.txt](https://github.com/retrospect-addon/plugin.video.retrospect/files/8448635/category_items_changes.txt)

**Comment by basrieter** [comment]
So all except search should be fine now.

**Comment by basrieter** [comment]
Search fixed.

**Comment by matsarv** [comment]
Tested all fixes made above. Everything works perfectly again. Ready for release.

**Comment by basrieter** [comment]
Great, then I will merge this and an official release soon.

**Comment by matsarv** [comment]
I see you haven't made a pull request for the 5.5.5 release yet so so I've added it here. (If you want I can create a new issue)

When I search for a word they have now implemented that the search word is highlited in yellow.

<details><summary>SVT Play search images</summary>
<p>

#### Seaching for 'Halcyon' gets highlited in title
![svt-title](https://user-images.githubusercontent.com/46671821/162701030-a142f348-50e4-4f40-bb60-e2fb0c2b893c.png)

#### Seaching for 'Hamilton' gets highlited in description
![svt-desc](https://user-images.githubusercontent.com/46671821/162701081-5867c855-48bf-482b-b6e3-f77df49a0da5.png)
</p>
</details>

<details><summary>Retrospect search images</summary>
<p>

#### Seaching for 'Halcyon' gets emphasized in description but not in title
![retro-halcyon](https://user-images.githubusercontent.com/46671821/162702168-9315df78-d1fe-492c-afa8-ea6346674bc1.png)


#### Seaching for 'Hamilton' gets highlited in description
![retro-hamilton](https://user-images.githubusercontent.com/46671821/162702183-a2d20f06-076c-444c-b89c-6d4bcbda2cc8.png)


</p>
</details>


<details><summary>Code used for test</summary>
<p>

#### I've tried both HtmlHelper.to_text and using Kodi colorcode.

def create_api_search_hit

from

```
        # Clean up the name
        name = result_set["heading"]
        name = HtmlHelper.to_text(name)
        result_set["name"] = name
        result_set["heading"] = name
```

to

```
        # Clean up the name and description
        name = result_set["heading"]
        name = HtmlHelper.to_text(name)
        description = result_set["description"]
        #description = HtmlHelper.to_text(description)
        description = description.replace("<em>", "[COLOR=yellow]").replace("</em>", "[/COLOR]")
        result_set["name"] = name
        result_set["heading"] = name
        result_set["description"] = description
```

</p>
</details>


<details><summary>Result images</summary>
<p>

#### Maybe the search word in title should be highlited?

![retro-halcyon-fix](https://user-images.githubusercontent.com/46671821/162703008-aa773447-55d8-45e5-a8c3-9f827ded54bb.png)


</p>
</details>

**Comment by matsarv** [comment]
Updated the Code used and Result image

**Comment by basrieter** [comment]
That I can do! Let me check tonight.

**Comment by basrieter** [comment]
@matsarv Thanks! Could you verify?

**Comment by matsarv** [comment]
Verified, works well.

**Comment by matsarv** [comment]
All fixed - v5.5.5 released.
